src/components/dialogs/AddToSocialGroupDialog.jsx

```diff
@@ -0,0 +1,157 @@
+  const { data: siteSettings, loading: siteSettingsLoading } =
+    useSiteSettings();
+  const {
+    data: selectedGroupData,
+    loading: selectedGroupDataLoading,
+  } = useSocialGroup(selectedGroup);
+  const {
+    mutate: patchSocialGroup,
+    loading: patchLoading,
+    error: patchError,
+    clearError: clearPatchError,
+  } = usePatchSocialGroup();
+
+  const handleClose = useCallback(() => {
+    setSelectedGroup('');
+    setSelectedRole('');
+    if (patchError) clearPatchError();
+    onClose();
+  }, [patchError]);
+
+  const safeSocialGroups = socialGroups || [];
+  const roles = get(
+    siteSettings,
+    ['social_group_roles', 'value'],
+    [],
+  );
+
+  const formComplete = Boolean(selectedGroup && selectedRole);
+
+  return (
+    <StandardDialog
+      open={open}
+      onClose={handleClose}
+      titleId="ADD_TO_SOCIAL_GROUP"
+      maxWidth="xl"
+    >
+      <DialogContent>
+        <InputRow
+          loading={socialGroupsLoading}
+          schema={{ labelId: 'SOCIAL_GROUP' }}
+        >
+          <FormControl>
+            <InputLabel>
+              <FormattedMessage id="SOCIAL_GROUP" />
+            </InputLabel>
+            <Select
+              style={{ width: 200 }}
+              value={selectedGroup}
+              onChange={e => setSelectedGroup(e.target.value)}
+            >
+              {safeSocialGroups.map(group => (
+                <MenuItem key={group?.guid} value={group?.guid}>
+                  {group?.name}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+        </InputRow>
+        <InputRow
+          loading={siteSettingsLoading}
+          schema={{ labelId: 'ROLE' }}
+        >
+          <FormControl>
+            <InputLabel>
+              <FormattedMessage id="ROLE" />
+            </InputLabel>
+            <Select
+              style={{ width: 200 }}
+              value={selectedRole}
+              onChange={e => setSelectedRole(e.target.value)}
+            >
+              {roles.map(role => (
+                <MenuItem key={role?.guid} value={role?.guid}>
+                  {role?.label}
+                </MenuItem>
+              ))}
+            </Select>
+          </FormControl>
+        </InputRow>
+      </DialogContent>
+
+      <DialogActions
+        style={{
+          padding: '0px 24px 24px 24px',
+          display: 'flex',
+          alignItems: 'flex-end',
+          flexDirection: 'column',
+        }}
+      >
+        {patchError && (
+          <CustomAlert
+            style={{ margin: '20px 0', width: '100%' }}
+            severity="error"
+            titleId="SERVER_ERROR"
+          >
+            {patchError}
+          </CustomAlert>
+        )}
+        <div>
+          <Button
+            style={{ marginRight: 4 }}
+            onClick={handleClose}
+            id="CLOSE"
+          />
+          <Button
+            disabled={selectedGroupDataLoading || !formComplete}
```

Comment:
it should become enabled as soon as the information necessary to send the request is loaded. if we dont have the information yet, do we let them click the button and get an error?

Reply:
No, we don't show them the form until we have the information we need to process it.

Reply:
Alternatively, we could allow them to fill out the form if `selectedGroupDataLoading` is true, but allow them to submit and set this button to `loading=true`.

Reply:
I think it's okay to show the form elements while the necessary data is loading, and only allow the user to continue once the necessary data is there. We can always change this later if it feels weird but I'm not anticipating it will ever come up